### PR TITLE
Tail Entwining

### PR DESCRIPTION
**Pull request by Geevies**
* Ported tail entwining. Simply grab another person that has a tail to entwine tails with them. Make sure to do it in the dorm or risk a ban!

Ported from: https://github.com/tgstation/tgstation/pull/54991

A recreation of https://github.com/Aurorastation/Aurora.3/pull/10572, which ran into some technical issues.

A speedmerge is advised to prevent underhanded folks from foiling this PR.

### CODE REVIEW

**Review by Arrow768 — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 1]
```diff
@@ -199,7 +199,11 @@
 				G.last_action = world.time
 				visible_message("<span class='warning'>[M] gets a strong grip on [src]!</span>")
 				return 1
-			visible_message("<span class='warning'>[M] has grabbed [src] passively!</span>")
+			if(src.has_tail() && H.has_tail())
+				H.visible_message(SPAN_DANGER("[H] coils their tail with [src], wow is that okay in public?!"), "You entwine your tail with [src]'s!")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:203]:
```suggestion
				var/other_msg = SPAN_WARNING("[H] coils their tail with [src]'s.")
				if(length(viewers(src)|viewers(H)) > 3)
					other_msg = SPAN_DANGER("[H] coils their tail with [src]'s. Wow, is that okay in public?!")
				H.visible_message(other_msg, SPAN_NOTICE("You entwine your tail with [src]'s!"))
```

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 2]
```diff
@@ -199,7 +199,14 @@
 				G.last_action = world.time
 				visible_message("<span class='warning'>[M] gets a strong grip on [src]!</span>")
 				return 1
-			visible_message("<span class='warning'>[M] has grabbed [src] passively!</span>")
+			if(src.has_tail() && H.has_tail())
+				var/other_msg = SPAN_WARNING("[H] coils their tail with [src]'s.")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:203]:
```suggestion
```

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 3]
```diff
@@ -199,7 +199,14 @@
...
+				if(length(viewers(src)|viewers(H)) > 3)
+					other_msg = SPAN_DANGER("[H] coils their tail with [src]'s. Wow, is that okay in public?!")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:205]:
```suggestion
					H.visible_message(SPAN_DANGER("[H] coils their tail with [src]'s. Wow, is that okay in public?!"), SPAN_NOTICE("You entwine your tail with [src]'s!"))
					gib(H)
					return 1
```

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 4]
```diff
@@ -199,7 +199,14 @@
...
+				H.visible_message(other_msg, SPAN_NOTICE("You entwine your tail with [src]'s!"))
+				gib(H)
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:207]:
```suggestion
```

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 5]
```diff
@@ -199,7 +199,14 @@
 				G.last_action = world.time
 				visible_message("<span class='warning'>[M] gets a strong grip on [src]!</span>")
 				return 1
-			visible_message("<span class='warning'>[M] has grabbed [src] passively!</span>")
+			if(src.has_tail() && H.has_tail())
+				var/other_msg = SPAN_WARNING("[H] coils their tail with [src]'s.")
+				if(length(viewers(src)|viewers(H)) > 3)
+					other_msg = SPAN_DANGER("[H] coils their tail with [src]'s. Wow, is that okay in public?!")
+				H.visible_message(other_msg, SPAN_NOTICE("You entwine your tail with [src]'s!"))
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:206]:
```suggestion
				H.visible_message(SPAN_WARNING("[H] coils their tail with [src]'s."), SPAN_NOTICE("You entwine your tail with [src]'s!"))
```

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]

**Review by MarinaGryphon — APPROVED** [review]

**Comment by Arrow768** [comment]
Given the high priority of that PR and the potential performance improvements due to the reduced active char count in the game this will be speedmerged.

**Comment by KasparoVy** [comment]
Adorable!

**Pull request merged**